M365/src/DeleteContactsFromFolder/DeleteContactsFromFolder.ps1

```diff
@@ -0,0 +1,188 @@
+                $folderObj = $folders | Where-Object { $_.displayName -eq $folderToBeDeleted }
+            }
+
+            if ($null -ne $folderObj) {
+                # Get the contacts in the folder
+                $listContactsInFolder = @{
+                    Query       = "me/contactFolders/$($folderObj.id)/contacts?`$top=100"
+                    AccessToken = $token.AccessToken
+                    GraphApiUrl = $graphApiUrl
+                }
+                $contactsResponse = Invoke-GraphApiRequest @listContactsInFolder
+                if ($contactsResponse.Successful -eq $false) {
+                    Write-Host "Unable to query the contacts. Please try again." -ForegroundColor Red
+                    exit
+                }
+
+                # Get all the contacts in the folder by looping through the nextLink (pagination)
+                do {
+                    $contacts.AddRange($contactsResponse.Content.value)
+                    $query = $contactsResponse.Content.'@odata.nextLink'.replace("https://graph.microsoft.com/v1.0/", "")
+                    $listContactsInFolder.Query = $query
+                    $contactsResponse = Invoke-GraphApiRequest @listContactsInFolder
+                } until (-not($contactsResponse.Content.'@odata.nextLink'))
+
+                Write-Host "Number of contacts in the folder: '$($folderObj.displayName)' is: $($contacts.Count)"
+                $userConfirmation = Read-Host -Prompt "About to delete the folder and its content. Are you sure Y/N"
+
+                if ($userConfirmation.ToLower() -eq "y") {
+                    Write-Host "Deleting Folder contents..." -ForegroundColor Cyan
+                    # Loop through and delete the contacts
+                    foreach ($contact in $contacts) {
+                        Write-Host "Now processing: '$($contact.displayName)'" -ForegroundColor Cyan
+                        Write-Host "Id:  $($contact.id)" -ForegroundColor Cyan
+                        $deleteContactParams = @{
+                            Query              = "me/contactFolders/$($folderObj.id)/contacts/$($contact.id)"
+                            AccessToken        = $token.AccessToken
+                            Method             = "DELETE"
+                            ExpectedStatusCode = 204
+                            GraphApiUrl        = $graphApiUrl
+                        }
+                        $deleteContactResponse = Invoke-GraphApiRequest @deleteContactParams
+                        if ($deleteContactResponse.Successful -eq $false) {
+                            Write-Host "Unable to delete the contact" -ForegroundColor Red
+                            $unableToDeleteAllContacts = $true
+                        } else {
+                            Write-Host "Deleted contact successfully" -ForegroundColor Green
+                        }
+                    }
+
+                    if ($unableToDeleteAllContacts -eq $false) {
```

Comment:
Are we required to delete all the contacts prior to deleting a folder?

Reply:
This is needed in order to force the sync vs doing a folder delete. Should add this in the comments prior to deleting the contacts.

Reply:
Added a comment regarding this.